### DISCUSSION TITLE
Raw query with parameters

### DISCUSSION THREAD

**Discussion by MrFahrenheit-55** [serverpod/serverpod#1271] [Q&A] [answered]
Hi there, I am trying to write a raw query where tables are created from user input. How can I write the query dynamically to take a parameter for the table name?

**Comment by SandPod** [comment, marked as answer]
Hey! 👋 
Seems like [Executing raw queries](https://docs.serverpod.dev/concepts/database-communication#executing-raw-queries) can be used to achieve what you are after.

The query is defined as a String, which means you should be able to use string interpolation to define the query as you would like.

For example (not tested):
```
var table = 'MyAwesomeTable';
var result = await session.db.query('SELECT * FROM $table WHERE true');
```

**Reply by MrFahrenheit-55** [reply]
Thanks for the answer @SandPod. Would this not be vulnerable to sql injection?

**Reply by SandPod** [reply]
Yes, I would strongly discourage opening up the possibility for users to directly inject strings that are run on your database.

The answer above merely show how you can achieve what you asked for.

Have you explored the possibility to identify what model your users would like to interact with, then do your table modification through the ORM. Using the methods provided by the code generator?

**Reply by vlidholt** [reply]
I think we should probably add better support for this in the future (there may already be an issue to track it).

**Reply by MrFahrenheit-55** [reply]
Thanks, @SandPod. I have explored that possibility, but since I need to be able to create dimensions and facts based on user inputs, the model approach will not work. 

@vlidholt , that is excellent news, if you can maybe post the tracker link for me that would be fantastic!

**Reply by SandPod** [reply]
Just to fully understand your need, which best describes what you would need help with.

1. Sanitize the user input so that it becomes safe to use in a raw query towards the database.
2. Helper method for database operations to create tables and configure columns for those tables.

**Reply by MrFahrenheit-55** [reply]
I would say 1. is the most critical. 2. would be nice, but with raw queries, this is not as important as sanitizing the input

**Reply by SandPod** [reply]
Great, thanks for the clarification. I think 1. can be tracked with this issue given @vlidholt comment: https://github.com/serverpod/serverpod/issues/722
